app/models/author.rb

```diff
@@ -58,7 +45,10 @@ def end_date
 
   # @return [Array<Integer>] ScienceWireIds for approved publications
   def approved_sciencewire_ids
-    publications.approved.with_sciencewire_id.pluck(:sciencewire_id).uniq
+    publications.where("contributions.status = 'approved'")
+                .where.not(sciencewire_id: nil)
+                .pluck(:sciencewire_id)
+                .uniq
   end
```

Comment:
Why is this an improvement?

Reply:
Because it doesn't rely on the method defined in callback (using `arel_table`) and is just basic ActiveRecord.  The formatting is not the important part, if that is what you are responding to.